### PR TITLE
feat: add image name in session launcher

### PR DESCRIPTION
**Pull request by ggstone0523**
Resolve:  lablup/backend.ai-webui #1074 

- display session information in last page of session launcher

**test image**
PR1

### CODE REVIEW

File: src/components/backend-ai-session-launcher.ts [hunk 1]
```diff
@@ -3359,6 +3359,29 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
             </wl-expansion>
           </div>
           <div id="progress-04" class="progress center layout fade">
```
Comment (lizable) [inline src/components/backend-ai-session-launcher.ts:3391]: Let's add the type of session (`BATCH` or `INTERACTIVE`).
And I cannot see proper session info when I enabled two keys `debug` or `allowManualImageNameForSession` to true in configuration file(`config.toml`). Please update when the user uses the manual image.

**Screenshot**:
using-manual-image-in-session-launcher-01
using-manual-image-in-session-launcher-02

As you can see the pictures above, there's no image description and the environment doesn't match with logo.

**Review by lizable — CHANGES_REQUESTED** [review]
The layout seems fine, but why don't we enhance it more according to the comments below?

**Comment by ggstone0523** [comment]
### The revision was completed by reflecting the comments.

**1. add type of session**

**Interactive**
PR1_review1_1

**Batch**
PR1_review1_2


**2. display session information for manual image**

PR1_review1_3

**Review by lizable — CHANGES_REQUESTED** [review]
I would like to suggest two things:
First, Let's display session type below environment.
Although session type is important enough, but environment is much more important than the type.
And Also, How about making own property (Object) for image?
For now there are many duplicated string processing methods such as `value.split(':').pop()?.split('-')...`
Unless the purpose of the duplicated statement is for readability, converting to property and add general processing function would be better, I think.

**Comment by ggstone0523** [comment]
### The revision was completed by reflecting the comments.

**1. Display session type below environment**
PR1_review2_1

**2.  Change duplicated string processing methods to function**
- Change duplicated string processing methods for 'manualImageName' to '_preProcessingManualImageInfo' function.
- '_preProcessingManualImageInfo' function takes derive kernel infomation from manualImageName and save it in manualImageInfo.

File: src/components/backend-ai-session-launcher.ts [hunk 2]
```diff
@@ -191,6 +191,14 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
   @property({type: Object}) _boundFolderMapRenderer = this.folderMapRenderer.bind(this);
   @property({type: Boolean}) useScheduledTime = false;
   @property({type: Object}) schedulerTimer;
+  @property({type: Object}) sessionInfoObj = {
+    'selectorEnvironment': '',
+    'selectorVersion': ['', ''],
+    'isSelectorTrue': false,
```
Comment (agatha197) [inline src/components/backend-ai-session-launcher.ts]: How about renaming this variable to `isVersionSelected`?

**Comment by ggstone0523** [comment]
### The revision was completed by reflecting the comments.

**1. Display session type below session environment**
PR1_review3_1

**2. Change repeated element to object**
- Change repeated element in 'sessionInfo' to 'sessionInfoObj' object
- '_preProcessingSessionInfo' function takes derive kernel infomation from manualImageName or selector and save it in 'sessionInfoObj'.
- '_processingSessionEnvironment' function is processing session environment value.
- '_processingSessionVersion' function is processing session version value.

File: src/components/backend-ai-session-launcher.ts [hunk 3]
```diff
@@ -191,6 +191,14 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
+    'manualEnvironment': '',
+    'manualVersion': ['', ''],
+    'isManualTrue': false
```
Comment (agatha197) [inline src/components/backend-ai-session-launcher.ts]: I think `isManualImageName` would be more comprehensive.

File: src/components/backend-ai-session-launcher.ts [hunk 4]
```diff
@@ -1033,6 +1041,54 @@ export default class BackendAiSessionLauncher extends BackendAIPage {
...
+      this.sessionInfoObj.isSelectorTrue = true;
+    } else {
+      this.sessionInfoObj.isSelectorTrue = false;
+    }
+
+    if (this.manualImageName?.value) {
+      this.sessionInfoObj.isManualTrue = true;
+    } else {
+      this.sessionInfoObj.isManualTrue = false;
+    }
```
Comment (agatha197) [inline src/components/backend-ai-session-launcher.ts]: This code is same with `this.sessionInfoObj.isManualTrue = this.manualImageName?.value ? true : false`. But it's okay now, so you can go in a better direction as you think.

**Review by Sujin-Kim1 — CHANGES_REQUESTED** [review]

**Comment by ggstone0523** [comment]
### The revision was completed by reflecting the comments.

**1. Change sessionInfoObj attribute name**
- change 'isSelectorTrue' to 'isVersionSelected'
- change 'isManualTrue' to 'isManualImageName'

**2. Change _preProcessingSessionInfo function logic**
- change if statement to assignment statement

**Comment by ggstone0523** [comment]
### The revision was completed.

**1. display icon in session info for non installed manual image**
**2. Separate Constraint infomation from Base information**

**Review by lizable — APPROVED** [review]
LGTM.

**Review by Sujin-Kim1 — APPROVED** [review]
LGTM

**Review by inureyes — APPROVED** [review]
LGTM.

**Pull request merged**